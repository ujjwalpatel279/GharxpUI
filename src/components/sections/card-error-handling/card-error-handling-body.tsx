import React, { FunctionComponent, ReactElement } from 'react';
import { CardBody } from '../../shared/card/card-body';

import './card-error-handling.scss';

export const CardErrorHandlingBody: FunctionComponent = (): ReactElement => {
  return (
    <CardBody>
      <div className="s-card-bad-redirects__detail">
        <div className="s-card-bad-redirects__detail__content">
          <b>Something went wrong !</b>
        </div>
      </div>
      <div className="s-card-bad-redirects__detail">
        <div className="c-cart__detail__content">
          <p>something went wrong and we couldn&apos;t complete your request. please refresh the page to try again</p>{' '}
        </div>
      </div>
    </CardBody>
  );
};
