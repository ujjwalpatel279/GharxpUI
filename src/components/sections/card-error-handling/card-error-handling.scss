@import '~sassHelpers';

.full-width {
  width: 100%;
}

.c-card-bad-redirects__header {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  .c-icon {
    fill: #ffffff;
  }
}
.c-card-bad-redirects__header__top {
  border-bottom: 0;
}
footer {
  text-align: center;
}
.c-card-bad-redirects {
  &__address {
    display: flex;
    width: 100%;
    padding: spacer(2) 0;
    border-bottom: 1px solid color(grey4);

    &__icon {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 15%;
    }
    &__text {
      width: 85%;
      padding-left: spacer(2);
    }
  }
  &__body {
    display: flex;
    flex-direction: column;
    padding: spacer(3) 0;
    color: color(grey2);
  }
}

.s-card-bad-redirects {
  &__detail {
    display: flex;
    align-items: center;
    justify-content: space-around;
    padding: spacer(2) 0;
    border-bottom: 1px solid color(grey4);

    &__content {
      display: flex;
      align-items: center;
    }

    p {
      text-align: center;
    }

    b {
      text-align: center;
    }

    span {
      flex: 4;
      margin-right: spacer(4);
      font-size: small;
    }
  }
}
