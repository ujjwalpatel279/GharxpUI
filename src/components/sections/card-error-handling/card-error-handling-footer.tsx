import React, { FunctionComponent, ReactElement } from 'react';
import { useHistory } from 'react-router-dom';

import { CardFooter } from '../../shared/card/card-footer';
import { Button } from '../../shared/button/button';

import './card-error-handling.scss';
interface CardErrorHandlingFooterInterface {
  url?: string;
}
export const CardErrorHandlingFooter: FunctionComponent<CardErrorHandlingFooterInterface> = ({ url }): ReactElement => {
  const history = useHistory();

  const tryAgain = () => {
    if (url && url.length > 1) {
      history.push(url);
    } else {
      history.goBack;
    }
  };

  return (
    <CardFooter>
      <footer>
        <div>
          <Button type="button" variant="primary" childClasses="full-width" onClick={tryAgain}>
            Try Again
          </Button>
        </div>
      </footer>
    </CardFooter>
  );
};
