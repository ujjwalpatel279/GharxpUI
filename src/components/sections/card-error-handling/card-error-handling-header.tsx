import { Icon } from '../../shared';
import React, { FunctionComponent, ReactElement } from 'react';
import { CardHeader } from '../../shared/card/card-header';

import './card-error-handling.scss';

export const CardErrorHandlingHeader: FunctionComponent = (): ReactElement => {
  return (
    <CardHeader>
      <header className="c-card-bad-redirects__header">
        <div className="c-card-bad-redirects__header__top">
          <div className="c-card__header__avatar">
            <div className="c-card__header__avatar__image">
              <Icon name="applicants" />
            </div>
          </div>
        </div>
      </header>
    </CardHeader>
  );
};
