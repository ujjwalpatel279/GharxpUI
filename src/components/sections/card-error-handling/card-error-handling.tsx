import React, { FunctionComponent, ReactElement } from 'react';
import { Card } from '../../shared';
import { CardErrorHandlingBody } from './card-error-handling-body';
import { CardErrorHandlingFooter } from './card-error-handling-footer';
import { CardErrorHandlingHeader } from './card-error-handling-header';

interface CardErrorHandlingInterface {
  url?: string;
}

export const CardErrorHandling: FunctionComponent<CardErrorHandlingInterface> = ({ url }): ReactElement => {
  return (
    <Card>
      <CardErrorHandlingHeader />
      <CardErrorHandlingBody />
      <CardErrorHandlingFooter url={url} />
    </Card>
  );
};
