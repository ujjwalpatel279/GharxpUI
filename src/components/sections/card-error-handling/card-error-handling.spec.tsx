import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { CardErrorHandling } from './card-error-handling';

describe('shows card bad redirects', () => {
  afterEach(cleanup);
  it('show bad redirects information on card', () => {
    const { container } = render(<CardErrorHandling />);
    expect(container).toMatchSnapshot();
  });
});
