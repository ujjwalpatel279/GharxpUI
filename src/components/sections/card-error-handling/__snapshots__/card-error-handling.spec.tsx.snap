// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows card bad redirects show bad redirects information on card 1`] = `
<div>
  <div
    class="c-card u-mb-3"
  >
    <div
      class="c-card__header"
    >
      <header
        class="c-card-bad-redirects__header"
      >
        <div
          class="c-card-bad-redirects__header__top"
        >
          <div
            class="c-card__header__avatar"
          >
            <div
              class="c-card__header__avatar__image"
            >
              <svg
                class="c-icon"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 4.272c1.467 0 2.656 1.206 2.656 2.694 0 0.803-0.347 1.525-0.897 2.018 1.36 0.565 1.863 1.825 2.046 3.129l0.025 0.191c0.011 0.096 0.021 0.191 0.029 0.287l0.015 0.239-7.752 0.014 0.007-0.098 0.016-0.198 0.020-0.198 0.011-0.097 0.026-0.193c0.189-1.286 0.696-2.52 2.038-3.077-0.55-0.493-0.897-1.214-0.897-2.018 0-1.488 1.189-2.694 2.656-2.694zM11.903 3.156c1.538 0 2.785 1.206 2.785 2.694 0 0.803-0.364 1.525-0.941 2.018 1.425 0.564 1.952 1.822 2.145 3.125l0.026 0.191c0.008 0.064 0.015 0.127 0.021 0.191l0.018 0.2 0.017 0.248 0.012 0.246 0.009 0.291 0.004 0.378 0 0.092-2.943 0c0.003-0.067 0.004-0.134 0.004-0.202 0-1.579-0.632-3.12-1.768-4.163 0.237-0.483 0.37-1.023 0.37-1.593 0-1.25-0.638-2.356-1.617-3.030 0.492-0.427 1.143-0.686 1.857-0.686zM4.087 3.156c0.718 0 1.373 0.264 1.866 0.698-0.967 0.674-1.597 1.775-1.597 3.018 0 0.57 0.133 1.111 0.369 1.594-1.133 1.043-1.763 2.583-1.763 4.163 0 0.068 0.001 0.135 0.004 0.202l-2.966-0 0.001-0.237 0.003-0.188 0.004-0.191 0.007-0.193 0.010-0.195 0.013-0.197 0.008-0.099 0.019-0.198c0.017-0.161 0.038-0.323 0.065-0.483l0.035-0.192c0.232-1.18 0.783-2.275 2.082-2.791-0.575-0.493-0.938-1.214-0.938-2.018 0-1.488 1.244-2.694 2.779-2.694z"
                />
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="c-card__body"
    >
      <div
        class="s-card-bad-redirects__detail"
      >
        <div
          class="s-card-bad-redirects__detail__content"
        >
          <b>
            Something went wrong !
          </b>
        </div>
      </div>
      <div
        class="s-card-bad-redirects__detail"
      >
        <div
          class="c-cart__detail__content"
        >
          <p>
            something went wrong and we couldn't complete your request. please refresh the page to try again
          </p>
           
        </div>
      </div>
    </div>
    <div
      class="c-card__footer "
    >
      <footer>
        <div>
          <button
            class="c-btn c-btn--primary full-width"
            type="button"
          >
            <span>
              Try Again
            </span>
          </button>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
